src/main/AuthenticationResponse.ts

```diff
@@ -256,56 +231,66 @@ function extractPresentations(resOpts: AuthenticationResponseOpts) {
     sst.includes(SubjectSyntaxTypesSupportedValues.DID.valueOf())
   );
-
   const state = resOpts.state || getState(verifiedJwt.payload.state);
   const nonce = verifiedJwt.payload.nonce || resOpts.nonce || getNonce(state);
-  const registration = createDiscoveryMetadataPayload(resOpts.registration);
-
-  //https://sphereon.atlassian.net/browse/VDX-140
-  // *********************************************************************************
-  // todo We are missing a wrapper object. Actually the current object is the id_token
-  // *********************************************************************************
-
-  const { verifiable_presentations, vp_token } = extractPresentations(resOpts);
-  const authenticationResponsePayload: AuthenticationResponsePayload = {
+  const idToken: IdToken = {
     iss: ResponseIss.SELF_ISSUED_V2,
-    sub: resOpts.did,
     aud: verifiedJwt.payload.redirect_uri,
-    did: resOpts.did,
-    sub_type: supportedDidMethods.length && resOpts.did ? SubjectIdentifierType.DID : SubjectIdentifierType.JKT,
-    state,
-    nonce,
     iat: Date.now() / 1000,
     exp: Date.now() / 1000 + (resOpts.expiresIn || 600),
-    registration,
+    sub: resOpts.did,
+    auth_time: Date.now() / 1000,
+    nonce,
+    _vp_token: {
+      // right now pex doesn't support presentation_submission for multiple VPs (as well as multiple VPs)
+      presentation_submission: {
+        id: uuidv4(),
+        definition_id: verifiedJwt.presentationDefinitions[0]?.definition.id || undefined,
+        descriptor_map: [
+          {
+            format:
+              resOpts.vp && ObjectUtils.isString(resOpts.vp[0]) ? VerifiablePresentationTypeFormat.JWT_VP : VerifiablePresentationTypeFormat.LDP_VP,
+            path: '$',
+          },
+        ],
+      } as PresentationSubmission,
+    },
+  };
+  if (supportedDidMethods.indexOf(SubjectSyntaxTypesSupportedValues.JWK_THUMBPRINT) != -1 && !resOpts.did) {
+    const { thumbprint, subJwk } = await createThumbprintAndJWK(resOpts);
+    idToken['sub_jwk'] = subJwk;
+    idToken.sub = thumbprint;
+  }
+  return idToken;
+}
+
+async function createSIOPResponsePayload(
+  verifiedJwt: VerifiedAuthenticationRequestWithJWT,
+  resOpts: AuthenticationResponseOpts
+): Promise<AuthenticationResponsePayload> {
+  assertValidResponseOpts(resOpts);
+  if (!verifiedJwt || !verifiedJwt.jwt) {
+    throw new Error(SIOPErrors.VERIFY_BAD_PARAMS);
+  }
+  const state = resOpts.state || getState(verifiedJwt.payload.state);
+  const { vp_token } = extractPresentations(resOpts);
+  const authenticationResponsePayload: Partial<AuthenticationResponsePayload> = {
+    access_token: resOpts.accessToken,
+    token_type: resOpts.tokenType,
+    refresh_token: resOpts.refreshToken,
+    expires_in: resOpts.expiresIn,
+    state,
   };
   // add support for multiple VPs (VDX-158)
   if (resOpts.vp && resOpts.vp[0].location === PresentationLocation.ID_TOKEN) {
```

Comment:
These options should result in the VP token to either be part of the id token, or to be a sibling of it. Current implementation does not do that anymore

Reply:
Fixed